src/dkpy/uncertainty_structure.py

```diff
@@ -0,0 +1,357 @@
+class ComplexDiagonalBlock(UncertaintyBlock):
+    """Complex-valued diagonal uncertainty block."""
+
+    def __init__(self, num_channels: int):
+        """Instantiate :class:`ComplexDiagonalBlock`.
+
+        Parameters
+        ----------
+        num_channels : int
+            Number of inputs/outputs to the uncertainty block.
+
+        Raises
+        ------
+        ValueError
+            If ``num_channels`` is not greater than zero.
+        """
+        if num_channels <= 0:
+            raise ValueError("`num_channels` must be greater than 0.")
+
+        self._num_channels = num_channels
+
+    @property
+    def num_inputs(self) -> int:
+        return self._num_channels
+
+    @property
+    def num_outputs(self) -> int:
+        """Get number of output of the uncertainty block."""
+        return self._num_channels
+
+    @property
+    def is_diagonal(self) -> bool:
+        """Get boolean for diagonal uncertainty."""
+        return True
+
+    @property
+    def is_complex(self) -> bool:
+        """Get boolean for complex uncertainty."""
+        return True
+
+    @property
+    def is_square(self) -> bool:
+        """Get boolean for square uncertainty."""
+        return True
+
+
+class ComplexFullBlock(UncertaintyBlock):
+    """Complex-valued full uncertainty block."""
+
+    def __init__(self, num_inputs: int, num_outputs: int):
+        """Instantiate :class:`ComplexFullBlock`.
+
+        Parameters
+        ----------
+        num_inputs : int
+            Number of inputs (columns) to the uncertainty block.
+        num_outputs : int
+            Number of outputs (rows) to the uncertainty block.
+
+        Raises
+        ------
+        ValueError
+            If ``num_inputs`` is not greater than zero.
+        ValueError
+            If ``num_outputs`` is not greater than zero.
+        """
+
+        if num_inputs <= 0:
+            raise ValueError("`num_inputs` must be greater than 0.")
+        if num_outputs <= 0:
+            raise ValueError("`num_outputs` must be greater than 0.")
+
+        self._num_inputs = num_inputs
+        self._num_outputs = num_outputs
+
+    @property
+    def num_inputs(self) -> int:
+        return self._num_inputs
+
+    @property
+    def num_outputs(self) -> int:
+        """Get number of output of the uncertainty block."""
+        return self._num_outputs
+
+    @property
+    def is_diagonal(self) -> bool:
+        """Get boolean for diagonal uncertainty."""
+        return False
+
+    @property
+    def is_complex(self) -> bool:
+        """Get boolean for complex uncertainty."""
+        return True
+
+    @property
+    def is_square(self) -> bool:
+        """Get boolean for square uncertainty."""
+        return self._num_inputs == self._num_outputs
+
+
+class UncertaintyBlockStructure:
+    def __init__(self, block_structure):
+        self.block_list = self.convert_block_structure_representation(block_structure)
+
+    def convert_block_structure_representation(
```

Comment:
Since this does not reference `self` at all, it can be made a private helper function in this module, or a static method if you prefer it to be grouped here.